core/src/test/scala/unit/kafka/server/ReplicationQuotaTest.scala

```diff
@@ -0,0 +1,211 @@
+    TestUtils.createTopic(zkUtils, topic, Map(0 -> Seq(0, 1), 1 -> Seq(0, 1), 2 -> Seq(0, 1), 3 -> Seq(0, 1)), brokers)
+
+    //Define test settings
+    val msg = msg100KB
+    val throttle: Int = 10 * msg.length
+    val msgCount: Int = 50
+
+    //Set the throttle config and replicas list so partition 0 & 2, only, are throttled
+    changeBrokerConfig(zkUtils, (0 until brokers.length), property(KafkaConfig.ThrottledReplicationRateLimitProp, throttle.toString))
+    val side: Int = if (throttleSide == FollowerReplication) 1 else 0
+    changeTopicConfig(zkUtils, topic, property(ThrottledReplicasListProp, s"0-$side:2-$side")) //partition 0 & 2 are throttled
+
+    //Useful functions
+    def configPropagatedCondition(): Boolean = brokers(side).quotaManagers.leaderReplication.isThrottled(new TopicAndPartition(topic, 0)) //doesn't matter if we check leader or follower replication
+    def logsMatchRegular() = waitForOffset(tp(topic, 1), msgCount) && waitForOffset(tp(topic, 3), msgCount)
+    def logsMatchThrottled() = waitForOffset(tp(topic, 0), msgCount) && waitForOffset(tp(topic, 2), msgCount) //the throttled ones
+
+    //Wait for config to propagate
+    waitUntilTrue(configPropagatedCondition, "Throttled partition config should have propagated.")
+
+    val start: Long = System.currentTimeMillis()
+
+    //Write a message to each partition individually to get an even spread
+    for (x <- 0 until msgCount)
+      for (partition <- (0 to 3))
+        producer.send(new ProducerRecord(topic, partition, null, msg))
+
+    waitUntilTrue(logsMatchRegular, "Partition 1 or 3's logs didn't match", 30000)
+
+    var took = System.currentTimeMillis() - start
+    assertTrue("Partition 1 & 3 should have replicated quickly: " + took, took < 2000)
+
+    waitUntilTrue(logsMatchThrottled, "Throttled partitions (0,2) logs didn't match")
+
+    val expectedDuration = msgCount / (throttle / msg.length) * 1000 * 2 // i.e 2 throttled partitions
+    took = System.currentTimeMillis() - start
+    info(s"expected: $expectedDuration, was: $took")
+    assertEquals(s"Throttled partitions should have been slow. Was $took ms", expectedDuration, took, FifteenPercentError(expectedDuration))
+  }
+
+  def tp(topic: String, partition: Int): TopicAndPartition = new TopicAndPartition(topic, partition)
+
+  def logsMatch(): Boolean = logsMatch(TopicAndPartition(topic1, 0))
+
+  def logsMatch(topicAndPart: TopicAndPartition): Boolean = {
+    var result = true
+    val expectedOffset = brokers.head.getLogManager().getLog(topicAndPart).get.logEndOffset
+    result = result && expectedOffset > 0 && brokers.forall { item =>
+      expectedOffset == item.getLogManager().getLog(topicAndPart).get.logEndOffset
+    }
+    if (result) info("final offset was " + expectedOffset + " for partition " + topicAndPart)
+    result
+  }
+
+  def waitForOffset(topicAndPart: TopicAndPartition, offset: Int, servers: Seq[KafkaServer] = brokers): Boolean = {
+    var result = true
```

Comment:
It most certainly is. Thank you. I'll do a refactor of these tests in my final pass.